### PR TITLE
Itc improvements

### PR DESCRIPTION
**Pull request by cquiroz**
This PR contains two improvements:
* The title bar for ITC while now show when we are retrieving data or in an error
* Move the calculation of the wavelength at max to the backend
Pending on 
https://github.com/gemini-hlsw/lucuma-itc/pull/310

### CODE REVIEW

File: explore/src/main/scala/explore/itc/ItcPanelTitle.scala [hunk 1]
```diff
@@ -60,53 +69,77 @@ object ItcPanelTitle:
...
           ExploreStyles.ItcTileTitle,
           <.label(s"Target:"),
           Dropdown(
             clazz = ExploreStyles.ItcTileTargetSelector,
-            value = props.selectedTarget.get.orUndefined,
+            value = idx,
             onChange = {
-              case t @ ItcTarget(_, _, _) => props.selectedTarget.set(newSelected(t))
-              case _                      => Callback.empty
+              case t: Int => props.selectedTarget.set(newSelected(t))
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/itc/ItcPanelTitle.scala:135]: Why was it necessary to switch to value to an index? Just to know if this is something to consider in the future.

**Review by rpiaggio — APPROVED** [review]
LGTM

File: explore/src/main/scala/explore/itc/ItcPanelTitle.scala (hunk 1 above)
Comment (cquiroz) [reply explore/src/main/scala/explore/itc/ItcPanelTitle.scala:135]: Right, this was tricky.
The object set in value is compared (I think) internally as js objects which make it not match as you'd expect a scala object .
We may want to have a `Droopdown[A]` eventually

**Review by toddburnside — APPROVED** [review]
LGTM

File: explore/src/main/scala/explore/itc/ItcPanelTitle.scala (hunk 1 above)
Comment (toddburnside) [reply explore/src/main/scala/explore/itc/ItcPanelTitle.scala:135]: I have already changed it to `Dropdown[A]` in my recent work to make it safer. There will be a `Dropdown[A]` and a `DropdownOptional[A]`

Comment (cquiroz) [reply explore/src/main/scala/explore/itc/ItcPanelTitle.scala:135]: Oh great, would it it take an `Eq[A]`?

Comment (toddburnside) [reply explore/src/main/scala/explore/itc/ItcPanelTitle.scala:135]: When I upgrade explore to the new lucuma-react, I'll see if this works correctly.

Comment (toddburnside) [reply explore/src/main/scala/explore/itc/ItcPanelTitle.scala:135]: > Oh great, would it it take an `Eq[A]`?

Hmm. It doesn't at the moment. So, the problem you are having is that the correct `SelectItem` isn't selected for the target?

**Pull request merged**